skymap_scanner/recos/__init__.py

```diff
@@ -31,6 +37,40 @@ class RecoInterface:
     # The spline files will be looked up in pre-defined local paths or fetched from a remote data store.
     SPLINE_REQUIREMENTS: List[str] = list()
 
+    def init(self):
+        raise NotImplementedError()
+
+    @staticmethod
+    def get_vertex_variations() -> List[I3Position]:
+        """Returns a list of vectors referenced to the origin that will be used to generate the vertex position variation."""
+        raise NotImplementedError()
+
+    @staticmethod
+    def do_rotate_vertex() -> bool:
+        """Defines whether each generated vertex variation should be rotated along the axis of the scan direction. With the exception for legacy algorithms (MillipedeOriginal) this should typycally return True."""
+        return True
+
+    @staticmethod
+    def do_refine_time() -> bool:
+        """Defines whether to refine seed time."""
+        return True
```

Comment:
Since these are pure booleans I am considering moving them to a dictionary with default values, from which individual recos can deviate with their own config in `__init__`.

Now we basically use these mostly/only to support the "legacy" configuration of `millipede_original`.

This would also make more straightforward to configure them at runtime.

Reply:
Dictionary is too free-form. Turned `RecoInterface` into an abstract class and declared annotated attributes.